rust/src/transformations/quantile_score_candidates/pseudocode/make_quantile_score_candidates.py

```diff
@@ -0,0 +1,45 @@
+# type: ignore
+def make_quantile_score_candidates(
+    input_domain: VectorDomain[AtomDomain[TIA]],
+    input_metric: MI,
+    candidates: List[TIA],
+    alpha: A
+) -> Transformation:
+
+    input_domain.element_domain.assert_non_null()
+
+    for i in range(len(candidates) - 1):
+        assert candidates[i] < candidates[i + 1]
+
+    alpha_numer, alpha_denom = alpha.into_frac(size=None)
+    if alpha_numer > alpha_denom or alpha_denom == 0:
+        raise ValueError("alpha must be within [0, 1]")
+
+    if input_domain.size is not None:
+        # to ensure that the function will not overflow
+        input_domain.size.inf_mul(alpha_denom)
+        size_limit = input_domain.size
+    else:
+        size_limit = (usize.MAX).neg_inf_div(alpha_den)
+
+    def function(arg: List[TIA]) -> List[usize]:
+        return compute_score(arg, candidates, alpha_numer, alpha_denom, size_limit)
+
+    if input_domain.size is not None:
+        def stability_map(d_in: u32) -> usize:
+            return TOA.inf_cast(d_in // 2).inf_mul(4).inf_mul(alpha_denom)
+    else:
+        abs_dist_const: usize = max(alpha_numer, alpha_denom.inf_sub(alpha_numer))
+        sup_dist_const: usize = abs_dist_const.inf_mul(2)
+        stability_map = new_stability_map_from_constant(sup_dist_const, QO=usize)
```

Comment:
2 · d_in · max(αnum, αden − αnum)

multiply by d_in missing?

Reply:
The d_in isn't missing there, it's just hidden! `new_stability_map_from_constant(C)` gives you a stability map of `d_in * C`

Reply:
yes, makes sense, thanks